stock_secondary_unit/models/stock_move.py

```diff
@@ -4,31 +4,49 @@
 from odoo.tools.float_utils import float_round
 
 
-class StockSecondaryUnitMixin(models.AbstractModel):
-    _name = "stock.secondary.unit.mixin"
-    _description = "Stock Secondary Unit Mixin"
+class StockMove(models.Model):
+    _inherit = ["stock.move", "product.secondary.unit.mixin"]
+    _name = "stock.move"
+    _secondary_unit_fields = {
+        "qty_field": "product_uom_qty",
+        "uom_field": "product_uom",
+    }
 
-    secondary_uom_id = fields.Many2one(
-        comodel_name="product.secondary.unit", string="Second unit"
-    )
-    secondary_uom_qty = fields.Float(
-        string="Secondary Qty", digits="Product Unit of Measure"
+    secondary_uom_qty = fields.Float()
+    secondary_uom_id = fields.Many2one()
+    product_uom_qty = fields.Float(
+        store=True, readonly=False, compute="_compute_product_uom_qty"
     )
 
+    @api.depends("secondary_uom_qty", "secondary_uom_id")
+    def _compute_product_uom_qty(self):
+        self._compute_helper_target_field_qty()
 
-class StockMove(models.Model):
-    _inherit = ["stock.move", "stock.secondary.unit.mixin"]
-    _name = "stock.move"
+    @api.onchange("product_uom")
+    def onchange_product_uom_for_secondary(self):
+        self._onchange_helper_product_uom_for_secondary()
 
     def _merge_moves_fields(self):
         res = super()._merge_moves_fields()
-        res["secondary_uom_qty"] = self[-1:].secondary_uom_qty
+        res["secondary_uom_qty"] = sum(self.mapped("secondary_uom_qty"))
         return res
 
+    @api.model
+    def _prepare_merge_moves_distinct_fields(self):
+        # Not merge moves with distinct secondary units
```

Comment:
You should put this as a docstring:
```suggestion
        """Don't merge moves with distinct secondary units."""
```